### DISCUSSION TITLE
can't connect to kerberized CDP using beeline

### DISCUSSION THREAD

**Discussion by damnworld** [apache/kyuubi#5385] [Q&A]
I am trying to submit Spark SQL to the CDP cluster through kyuubi, which is managed by CM. The cluster has Kerberos enabled. Among them, node n1/n2/n3 uses the kyuubi user to start the kyuubi server, and node c1(192.168.23.100) connects to the kyuubi server as a client. 
1. Log in successfully through the keytab of c1. 
2. Create a connection using beeline
`beeline -u "jdbc:hive2://n2:10009/default;principal=kyuubi/n2@DEV.CDP.COM"`
At this time, the error message returned is as follows
`Error: Could not open client transport with JDBC Uri: jdbc:hive2://n2:10009/default;principal=kyuubi/n2@DEV.CDP.COM: Error opening session for kyuubi client ip 192.168.23.100, due to Could not initialize class org.apache.kyuubi.engine.KyuubiApplicationManager$`

**Comment by pan3793** [comment]
> due to Could not initialize class org.apache.kyuubi.engine.KyuubiApplicationManager$

obviously, the message is incomplete, there should be more clues on the Kyuubi server's log

**Reply by damnworld** [reply]
![image](https://github.com/apache/kyuubi/assets/20886060/062ca15d-646d-45c3-9be9-ff6376c91f40)

**Reply by pan3793** [reply]
it indicates something wrong with initializing `KyuubiApplicationManager` class, e.g. running static code blocks. any more context you can provide? e.g. are you using the modified version of Kyuubi 1.7.1? have you put any jars under kyuubi classpath?

PS: please use text to provide the information as much as possible, the picture is not searchable, which is not friendly for future explorers

**Reply by damnworld** [reply]
Thanks a lot,@pan3793!
I just tried modifying the permissions of the user ‘kyuubi’ by adding full permissions to the database, and it connected successfully. Could you please tell me what are the requirements for user permission settings?

**Reply by pan3793** [reply]
I'm not sure what you mean "adding full permissions to the database"? Basically, should be same as user "hive" used by HiveServer2

**Reply by damnworld** [reply]
Got it,I gave user kyuubi the same permissions as user 'hive'.
